packages/server/lib/modes/run.js

```diff
@@ -1120,27 +1118,31 @@ module.exports = {
 
       obj.shouldUploadVideo = suv
 
-      debug('attempting to close the browser')
+      let videoCaptureFailed = false
+
+      if (endVideoCapture) {
+        await endVideoCapture()
+        .tapCatch(() => videoCaptureFailed = true)
+        .catch(warnVideoRecordingFailed)
+      }
 
       // always close the browser now as opposed to letting
       // it exit naturally with the parent process due to
       // electron bug in windows
-      return openProject
-      .closeBrowser()
-      .then(() => {
-        if (endVideoCapture) {
-          return this.postProcessRecording(
-            endVideoCapture,
-            videoName,
-            compressedVideoName,
-            videoCompression,
-            suv,
-          ).then(finish)
-          // TODO: add a catch here
```

Comment:
ah, `TODO` comments, the origin of all the world's best bugs

![nice](https://i.imgur.com/4wnHZzP.gif)